cmd/soroban-cli/src/commands/tx/edit.rs

```diff
@@ -0,0 +1,145 @@
+use std::{
+    env, fs,
+    io::{stdin, Cursor, IsTerminal, Write},
+    path::PathBuf,
+    process::{self, Stdio},
+};
+
+use base64::{engine::general_purpose::STANDARD as BASE64, Engine as _};
+use stellar_xdr::curr;
+
+use crate::{commands::global, print::Print};
+
+const DEFAULT_JSON: &str = r#"{
+  "tx": {
+    "tx": {
+      "source_account": "",
+      "fee": 100,
+      "seq_num": 0,
+      "cond": "none",
+      "memo": "none",
+      "operations": [],
+      "ext": "v0"
+    },
+    "signatures": []
+  }
+}
+"#;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+
+    #[error(transparent)]
+    StellarXdr(#[from] stellar_xdr::curr::Error),
+
+    #[error(transparent)]
+    SerdeJson(#[from] serde_json::Error),
+
+    #[error(transparent)]
+    Base64Decode(#[from] base64::DecodeError),
+
+    #[error("Editor returned non-zero status")]
+    EditorNonZeroStatus,
+}
+
+// Command to edit the transaction
+/// e.g. `stellar tx new manage-data --data-name hello --build-only | stellar tx edit`
+#[derive(Debug, clap::Parser, Clone, Default)]
+#[group(skip)]
+pub struct Cmd {}
+
+impl Cmd {
+    pub fn run(&self, global_args: &global::Args) -> Result<(), Error> {
+        let print = Print::new(global_args.quiet);
+        let json: String = if stdin().is_terminal() {
+            DEFAULT_JSON.to_string()
+        } else {
+            let mut input = String::new();
+            stdin().read_line(&mut input)?;
+            let input = input.trim();
+            xdr_to_json::<curr::TransactionEnvelope>(input)?
+        };
+
+        let path = tmp_file(&json)?;
```

Comment:
[nit] should use [tempfile](https://docs.rs/tempfile/latest/tempfile/fn.tempfile.html) to automatically cleanup files

Reply:
i don't want the file to be removed automatically, because i'll keep the file around in case there are errors.